### PR TITLE
Fixes and optimizations for carbonate

### PR DESCRIPTION
**Pull request by iksaif**
This pull requests contains all the patches that we currently have on top of carbonate. I'll be happy to split it if you only want some of the commits.

- Fix average elapsed time
- Fix environment variables
- Add the ability to sync a specific time range only
- Add the ability to fully overwrite data when needed
- Improve error messages

---
This change is [Reviewable](https://reviewable.io/reviews/graphite-project/carbonate/68)

### CODE REVIEW

File: carbonate/sync.py [hunk 1]
```diff
@@ -77,7 +77,8 @@ def heal_metric(source, dest, start_time=0, end_time=None):
                 # fill_archives' start and end are the opposite
                 # of what you'd expect
                 fill_archives(
-                    source, dest, startFrom=end_time, endAt=start_time)
+                    source, dest, startFrom=end_time, endAt=start_time
```
Comment (deniszh) [inline carbonate/sync.py]: I think you forgot comma at the end.

**Comment by deniszh** [comment]
@iksaif - could you please fix error above? missing comma in line 80 in carbonate/sync.py ?

**Comment by iksaif** [comment]
Sorry for the bad rebase. I'm unlikely to have access to my computer before à few weeks. I'll re-push when I do.

**Comment by deniszh** [comment]
Ah, everything is fine, I was blind at 01:00 AM. 
Sorry, @iksaif - will merge it soon if no objections from other members.

**Comment by deniszh** [comment]
Sorry,  your PR is broken by friendly merge. Could you please rebase one more time? Much appreciated!

**Comment by deniszh** [comment]
Fixed in https://github.com/graphite-project/carbonate/pull/70

**Pull request closed without merging**